src/utils/ProfileProvider.tsx

```diff
@@ -100,7 +86,7 @@ export default function ProfileProvider({ children }: { children: ReactNode }) {
       newLanguages: ProfileLanguage[],
       newRoles: ProfileRole[],
     ) => {
-      if (!userId) throw new Error('Fatal: user is not logged in!');
+      if (!auth?.userId) throw new Error('Fatal: user is not logged in!');
       if (profileData) throw new Error('Fatal: user already has a profile!');
 
```

Comment:
i just realized we're checking for `auth.userId` but not using it here...
should we add a check for `auth.userId === newProfile.user_id`?